api/types/server.go

```diff
@@ -485,16 +501,34 @@ func (s *ServerV2) openSSHEC2InstanceConnectEndpointNodeCheckAndSetDefaults() er
 	return nil
 }
 
+// serverNameForEICE returns the deterministic Server's name for an EICE instance.
+func serverNameForEICE(s *ServerV2) string {
+	awsAccountID := s.GetAWSAccountID()
+	awsInstanceID := s.GetAWSInstanceID()
+
+	if awsAccountID != "" && awsInstanceID != "" {
+		return fmt.Sprintf("%s-%s", awsAccountID, awsInstanceID)
+	}
+
+	return uuid.NewString()
```

Comment:
Should this be an error or an empty string checked by the caller, instead?

Reply:
This is only called in the CheckAndSetDefaults, which ends up checking if the Server has the Account and Instance ID.
So, even if we default to an uuid, it will fail afterwards because one (or both) of those values are missing.

I added the check anyway so that we can fail earlier.